R/table-formula.R

```diff
@@ -0,0 +1,56 @@
+#' @title TableFormula
+#' @docType class
+#' @description  A table formula of the model (Typically related to a \code{Quantity} such as a parameter)
+#' @field allPoints Returns all points defined in the table formulafor a \code{TableFormula} or \code{NULL}  otherwise (Read-Only).
+#' @field useDerivedValues Indicates whether table values should be derived during solving. the ODE system. Default value is \code{TRUE}
+#' @field xDimension The dimension in which the x values are defined (Read-Only).
+#' @name Formula
+#' @section Methods:
+#' \describe{
+#'   \item{ addPoint(x, y)}{Adds a point to the table, with x value in base unit for XDimension and y value in base unit for Dimension}
+#'   \item{ removePoint(x, y)}{ Remove the point having the same x and y from the table}
+#'   }
+#' @format NULL
+TableFormula <- R6::R6Class(
+  "TableFormula",
+  inherit = Formula,
+  active = list(
+    allPoints = function(value) {
+      if (missing(value)) {
+        return(toValuePoints(rClr::clrCall(self$ref, "AllPointsAsArray")))
+      } else {
+        private$throwPropertyIsReadonly("allPoints")
+      }
+    },
+    useDerivedValues = function(value) {
+      private$wrapProperty("UseDerivedValues", value)
+    },
+    xDimension = function(value) {
+      dim <- private$wrapReadOnlyProperty("XDimension", value)
+      rClr::clrGet(dim, "Name")

```

Comment:
We do not have an actual dimesnion object as wrapper. So I am getting the "Name" from the net object directly.